### ISSUE TITLE
Time field of new attendance session should be populated by default

### ISSUE THREAD

**Issue by Bwise1** [danmarsden/moodle-mod_attendance#638]
When adding new attendance session, as the Date field is being automatically populated by the Date of the Day, can the "From" Field of the Time to be populated by the current Time the session want to be added.
![moodle](https://user-images.githubusercontent.com/26597408/196618678-9d1dc743-c154-49db-a84d-599fb990d7af.png)

**Comment by danmarsden** [comment]
I do find 00 : 00 to be a weird default value myself, but I'm not convinced it makes sense to set the default value to the current time - usually teachers would create the sessions prior to them being used.

I think it does make sense to tidy up this selector - it's also difficult for people to create "all day" sessions as mentioned in #464 

feel free to help working on this by proposing a code-based solution.

**Comment by Bwise1** [comment]
the default value to the current time feature can be done only for add session, (don't know how easy it is to achieve, but I will take a look at the code base and see ). and if you are creating prior to the time, at least you are not starting from 00.00, you are starting from a time closer to the time you want to use the attendance. saves you more time.

**Comment by danmarsden** [comment]
thanks - let me know any further thoughts you have - I'm still not convinced it makes sense to use the current time as the default - eg - it's 20:07 right now, and if I was creating a session as a teacher, I'm more likely to be wanting to create one for tomorrow - or later in the week during the day. - there's little point in using the minute value and setting that to 07 as it would be really unlikely that a session would be set to start at 7 minutes past the hour and it's unlikely a teacher would be creating a session for 20:00 in some organisations who operate during the 0900-1700 time frame.

we could drop in a static "default" value in the admin settings - allowing an admin user to pre-select a time/hour for both the start and end time that makes more sense. What might be cool is to have some form of default duration too - so that if someone changes the start time, it auto-matically adjusts the end time using javascript - so if the majority of the time an organisation has 1hr sessions, the teacher sets the start time to 09:00 and the javascript auto-adjusts the end time to 10:00.. but... what I'd really want to do is completely replace that session date start/end selector and make it operate in a much more user-friendly way... of course that's going to take a bit more investment of time though.

**Comment by Sythelux** [comment]
> the default value to the current time feature can be done only for add session, (don't know how easy it is to achieve, but I will take a look at the code base and see ). and if you are creating prior to the time, at least you are not starting from 00.00, you are starting from a time closer to the time you want to use the attendance. saves you more time.

I like this. wouldn't it work if we can set a default length in the settings of like 45, 60, 90min and a list of closest start times e.g. 9:30, 11:15, 13:05 and the start time is using either the closest or the last rounded down.

this would also solve my issue in #754 

The typical full example for us would be: "the teacher comes in and does the intro and waits for late commers. They then do the attendance so it is for example now 9:35 the lecture usually goes from 9:30 to 11:15. The teacher creates a new entry (as I said in the other one we can't pre-populate it) the system already has 9:30 to 11:15 pre-selected and only has to set everyone to attending and only the missing students to unattending. Done. 

I the teacher wants to create the entry at 9:25 it would be cool to have then also choose the same correct time slot, but it would be ok, if the teacher is then just forced to create it 5min later, why would you do an attendance before the lecture starts anyway. (no one at our school does it. the teachers usually do attendance half way in, when there is time)

**Comment by danmarsden** [comment]
yes - there are a bunch of things that would be good here in that time selector.

1) the ability for the start time default to be set by the admin rather than always set to 0:00
2) The ability for some preset default duration to be defined, so that when a teacher changes from 9am to 10am the end time automatically changes to the duration after that point. This is what Google Calendar events do too - so if you set the start time to 9:15 - the end time auto changes to 10:15

The default "duration" could be 1hr which an admin user could potentially change - so if you wanted 45 min sessions by default you could do that.

Keep in mind that Attendance now also allows you open the session "early" so if students arrive 5 minutes before the session they can still start taking attendance for the session.

**Comment by danmarsden** [comment]
just copying the comment I made on #765 - I don't really like the complexity added by the patch in #765

what I'd like to see is:

1) A new site-level setting to define the default "start" time.
2) A new site level setting to define the default "duration"

And then some JavaScript that automatically updates the "end" time when someone changes the "start time" by the default duration amount.
